### ISSUE TITLE
AGW: Fixing S1AP test cases

### ISSUE THREAD

**Issue by VinashakAnkitAman** [magma/magma#5842]
### Your Environment

- **Version:** Local Development Setup
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** Magma MME/SPGW service
- **Deployment Environment:** Vagrant (AGW)

### Setup Details
The S1APTester ([https://github.com/facebookexperimental/S1APTester](https://github.com/facebookexperimental/S1APTester)) was used along with Magma EPC in Vagrant based setup to reproduce the issue.

### Describe the Issue
While adding the non-sanity test cases in the defs.mk file as part of PR #5840, many of the test cases were found failing on individual execution, some of them were failing or getting stuck while testing in set of test cases under target NON_SANITY_TESTS. 

The execution logs can be found here:
https://drive.google.com/drive/folders/1KjIO9YX87EkUcJXdKEMItBDnJ5md2TII?usp=sharing

As per the current understanding, the following test cases are found to be failing:
```
s1aptests/test_outoforder_erab_setup_rsp_default_bearer.py
s1aptests/test_agw_offload_idle_active_ue.py
s1aptests/test_agw_offload_mixed_idle_active_multiue.py
s1aptests/test_secondary_pdn_with_dedicated_bearer_multiple_services_restart.py
s1aptests/test_outoforder_erab_setup_rsp_dedicated_bearer.py
s1aptests/test_attach_detach_with_ipv6_pcscf_and_dns_addr.py
s1aptests/test_ipv4v6_secondary_pdn_rs_retransmit.py
s1aptests/test_ipv4v6_secondary_pdn_spgw_initiated_ded_bearer.py
s1aptests/test_ipv6_secondary_pdn_rs_retransmit.py
s1aptests/test_ipv4v6_secondary_pdn.py
s1aptests/test_ipv4v6_secondary_pdn_with_ded_bearer.py
s1aptests/test_ipv6_secondary_pdn_with_ded_bearer.py
s1aptests/test_ipv4v6_secondary_pdn_with_ded_bearer_multi_ue.py
s1aptests/test_attach_detach_with_corrupt_stateless_mme.py
s1aptests/test_ipv4v6_secondary_pdn_multi_ue.py
s1aptests/test_attach_detach_setsessionrules_tcp_data.py
s1aptests/test_stateless_multi_ue_mixedstate_mme_restart.py
s1aptests/test_attach_with_multiple_mme_restarts.py
```

The following test cases were successfully passing while ran individually, but found to be failing while ran in a set of test cases:
```
s1aptests/test_paging_with_mme_restart.py
s1aptests/test_standalone_pdn_conn_req_with_apn_correction.py
s1aptests/test_attach_esm_information_apn_correction.py
s1aptests/test_enb_partial_reset_multi_ue.py
s1aptests/test_attach_ul_udp_data_multi_ue.py
s1aptests/test_attach_dl_udp_data_multi_ue.py
s1aptests/test_attach_ul_tcp_data_multi_ue.py
s1aptests/test_attach_dl_tcp_data_multi_ue.py
s1aptests/test_attach_dl_ul_tcp_data_multi_ue.py
s1aptests/test_attach_and_mme_restart_loop_detach_and_mme_restart_loop_multi_ue.py
s1aptests/test_no_auth_response_with_mme_restart_reattach.py
s1aptests/test_no_identity_rsp_with_mme_restart.py
```

The following test cases mostly get stuck
```
s1aptests/test_attach_esm_info_timerexpiration_max_retries.py
```

One exceptional issue found with less occurrence that with following test case, MME was found to get stuck in restart loop for sometimes:
```
s1aptests/test_outoforder_erab_setup_rsp_default_bearer.py
s1aptests/test_3495_timer_for_default_bearer_with_mme_restart.py
s1aptests/test_3495_timer_for_dedicated_bearer_with_mme_restart.py
```

**To Reproduce**
1) Take the list of commented non-sanity test cases from PR #5840 and run them individually and under the target NON_SANITY_TESTS

**Comment by ardzoht** [comment]
Hi @VinashakAnkitAman, thanks for flagging this, which commit hash were you basing your tests on? Also for IPv4v6 related tests, @pruthvihebbani was looking at these and mentioned there's something related with MME receiving create session failure on secondary pdn, might be worth to take a look into this specific failure.

**Comment by VinashakAnkitAman** [comment]
> Hi @VinashakAnkitAman, thanks for flagging this, which commit hash were you basing your tests on? Also for IPv4v6 related tests, @pruthvihebbani was looking at these and mentioned there's something related with MME receiving create session failure on secondary pdn, might be worth to take a look into this specific failure.

I was working on the latest master. I checked with Pruthvi also regarding this issue and after discussion with her I came to know she was getting failure during the session creation. While for me that session creation was happening properly and the issue was in further step when test script was waiting for Router Advertisement and in the pipelined it has got an exception by the time. 

Although as part of the latest master, following IPv6 test cases have been observed to be working fine and updated in PR #5840 
s1aptests/test_attach_detach_with_ipv6_pcscf_and_dns_addr.py
s1aptests/test_ipv4v6_secondary_pdn_rs_retransmit.py
s1aptests/test_ipv4v6_secondary_pdn_spgw_initiated_ded_bearer.py
s1aptests/test_ipv6_secondary_pdn_rs_retransmit.py
s1aptests/test_ipv4v6_secondary_pdn.py
s1aptests/test_ipv4v6_secondary_pdn_multi_ue.py

For all the other test cases which were failing individually, either fixes have been provided as part of PRs #5931, #5986, #5988 or issues have been raised with issue Ids #5930, #5991, #5992, #5997, #5998
